src/pybind/mgr/dashboard/frontend/src/app/shared/directives/dynamic-input-combobox.directive.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  Directive,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  EventEmitter,
+  HostListener
+} from '@angular/core';
+import { ComboBoxItem } from '../models/combo-box.model';
+import { ComboBoxService } from '../services/combo-box.service';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+@Directive({
+  selector: '[cdDynamicInputCombobox]'
+})
+export class DynamicInputComboboxDirective implements OnInit, OnDestroy {
+  @Input() items: ComboBoxItem[];
+
+  @Output() updatedItems: EventEmitter<ComboBoxItem[]> = new EventEmitter();
+
+  private searchSubject: Subject<string> = new Subject();
+  private selectedItems: ComboBoxItem[] = [];
+
+  constructor(private combBoxService: ComboBoxService) {}
+
+  ngOnInit() {
+    this.searchSubject.pipe(debounceTime(300), distinctUntilChanged()).subscribe((searchString) => {
```

Comment:
use the debounce time constant here as well for the combo box service

Reply:
done